src/include/duckdb/storage/compression/alprd/alprd_compress.hpp

```diff
@@ -0,0 +1,290 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/storage/compression/alprd/alprd_compress.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/storage/compression/patas/patas.hpp"
+#include "duckdb/storage/compression/alprd/algorithm/alprd.hpp"
+#include "duckdb/function/compression_function.hpp"
+#include "duckdb/storage/compression/alprd/alprd_analyze.hpp"
+#include "duckdb/storage/compression/alprd/alprd_constants.hpp"
+#include "duckdb/storage/compression/alp/alp_constants.hpp"
+
+#include "duckdb/common/helper.hpp"
+#include "duckdb/common/limits.hpp"
+#include "duckdb/common/types/null_value.hpp"
+#include "duckdb/function/compression/compression.hpp"
+#include "duckdb/main/config.hpp"
+#include "duckdb/storage/buffer_manager.hpp"
+
+#include "duckdb/storage/table/column_data_checkpointer.hpp"
+#include "duckdb/storage/table/column_segment.hpp"
+#include "duckdb/common/operator/subtract.hpp"
+
+#include <functional>
+
+namespace duckdb {
+
+template <class T>
+struct AlpRDCompressionState : public CompressionState {
+
+public:
+	using EXACT_TYPE = typename FloatingToExact<T>::type;
+	explicit AlpRDCompressionState(ColumnDataCheckpointer &checkpointer, AlpRDAnalyzeState<T> *analyze_state)
+	    : checkpointer(checkpointer),
+	      function(checkpointer.GetCompressionFunction(CompressionType::COMPRESSION_ALPRD)) {
+
+		input_vector = vector<EXACT_TYPE>(AlpRDConstants::ALP_VECTOR_SIZE, 0);
+		raw_input_vector = vector<T>(AlpRDConstants::ALP_VECTOR_SIZE, 0);
+		vector_null_positions = vector<uint16_t>(AlpRDConstants::ALP_VECTOR_SIZE, 0);
+
+		//! State variables from the analyze step that are needed for compression
+		state.left_parts_dict_map = analyze_state->state.left_parts_dict_map;
```

Comment:
to avoid a copy we can also use `std::move(..)` here
The analyze state is thrown away after this anyways so we're free to take this memory

Reply:
Makes sense! We have added this